File: python/balance_parens.py
def balance_parens(str):
    no_more_paren = False
    bracket_is_open = False
    final_string = ""

    # Iterates through the string, and tracks whether the bracket is open or closed
    for i in range(0, len(str)):
        if str.find(")", i) == -1:
            no_more_paren = True
        if str[i] == "(" and no_more_paren == True:
            continue
        elif  str[i] == "(" and bracket_is_open == False:
            bracket_is_open = True
            final_string += "("
        elif str[i] == "(" and bracket_is_open == True:
            continue
        elif str[i] == ")" and bracket_is_open == False:
            continue
        elif str[i] == ")" and bracket_is_open == True:
            bracket_is_open = False
            final_string += ")"
        else:
            final_string += str[i]
    return final_string

File: js/balanceParens.js
balanceParens = (str) =>
{
    let bracketIsOpen = false;
    let noMoreParen = false;
    let finalString = "";
        
    // Iterates through the string, and tracks whether the bracket is open or closed
    for (let i = 0; i < str.length; i++){
        if (str.indexOf(")", i) == -1 ){
            noMoreParen = true;
        }
        if (str.charAt(i) == "(" && noMoreParen == true){
            continue;
        }else if (str.charAt(i) == "(" && bracketIsOpen == false){
            bracketIsOpen = true;
            finalString += "(";
        }else if (str.charAt(i) == "(" && bracketIsOpen == true){
            continue;
        }else if (str.charAt(i) == ")" && bracketIsOpen == false){
            continue;
        }else if (str.charAt(i) == ")" && bracketIsOpen == true){
            bracketIsOpen = false;
            finalString += ")";
        }else{
            finalString += str.charAt(i);
        }      
    }
    return finalString
    // console.log(str.lastIndexOf("("));
    // console.log(str.lastIndexOf(")"));

    


    // let numberOfOpen = [];
    // let numberOfClosed = [];

    // Create Arrays of closed & Open brackets
    // for (let i = 0; i < str.length; i++){
    //     if (str.charAt(i) == "("){
    //         numberOfOpen.push("(");
    //     }else if (str.charAt(i) == ")"){
    //         numberOfClosed.push(")")
    //     }
    // console.log(numberOfOpen);
    // console.log(numberOfClosed);
    
}
// balanceParens("abc(d)e(fgh))(i)j)k")
// console.log(balanceParens("abc(d)(ef(g(h))ij)k)lm()o)p"))
module.exports = { balanceParens }